main.py

```diff
@@ -119,24 +129,53 @@ def main():
-		print("\nObservations for given level: \n", observations_for_given_level)
-		print("\nPredictions for given level: \n", predictions_for_given_level)
-		print("\nObjects Recognized for given level: \n", object_recognitions_for_given_level)
-		print("\nInstinctive Reactions for given level: \n", instinctive_reactions_for_given_level)
-
-	print("---------------------------------------------------")
-	print("Final Observations \n", observations)
-	print("---------------------------------------------------")
-	print("Final Predictions \n", predictions)
-	print("---------------------------------------------------")
-	print("Final Instinctive Reactions \n", instinctive_reactions)
+		if observations_for_given_level:
+			print("\nObservations for given level: \n", observations_for_given_level)
+		if predictions_for_given_level:
+			print("\nPredictions for given level: \n", predictions_for_given_level)
+		if object_recognitions_for_given_level:
+			print("\nObjects Recognized for given level: \n", object_recognitions_for_given_level)
+		if instinctive_reactions_for_given_level:
+			print("\nInstinctive Reactions for given level: \n", instinctive_reactions_for_given_level)
+
+	if observations:
+		print("---------------------------------------------------")
+		print("Final Observations \n", observations)
+	if predictions:
+		print("---------------------------------------------------")
+		print("Final Predictions \n", predictions)
+	if instinctive_reactions:
+		print("---------------------------------------------------")
+		print("Final Instinctive Reactions \n", instinctive_reactions)
+	if object_recognized:
+		print("---------------------------------------------------")
+		print("Final Object Recognized \n", object_recognized)
 	print("---------------------------------------------------")
-	print("Final Object Recognized \n", object_recognized)
-	print("---------------------------------------------------")
-
-	print("Following are the new inputs that you entered and were not in the graph. Please add the necessary edges (Deductions/Inferences) for the input")
-	print(not_in_graph_inputs)
 
+	not_in_graph_inputs = list(not_in_graph_inputs)
+	nodes_added = {}
+	data = get_all_objects()
+	objects = data["objects"]
+	object_names = [preprocess_name(element["name"]) for element in objects]
+	for node in not_in_graph_inputs:
+		if node in object_names:
+			similar_objects = get_similar_elements(node)
+			if not similar_objects:
+				continue
+			most_similar_object = similar_objects[0]
+			most_similar_object_name = preprocess_name(most_similar_object[1]["name"])
+			graph.add_node(node)
+			successors = graph.successors(most_similar_object_name)
+			edge_attributes = {}
+			for successor in successors:
+				graph.add_edge(node, successor)
+				edge_attributes[(node, successor)] = graph.get_edge_data(most_similar_object_name, successor)
+			nx.set_edge_attributes(graph, edge_attributes)
+			nodes_added[node] = (most_similar_object_name, most_similar_object[0])
```

Comment:
What is this line doing?

Reply:
Line no. 174?
It is storing new nodes added and the similar object used to add edges and its similarity index